### PR TITLE
Adds independent layer controls to widget

### PR DESCRIPTION
**Pull request by K-Meech**
## Description

**What is this PR**

- [ ] Bug fix
- [X] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
Currently our widget contains no layer controls.

**What does this PR do?**
This PR adds a layer controls window to our widget, which responds to selection / insertion / deletion of group layers.

## References

Closes https://github.com/brainglobe/napari-experimental/issues/6

## How has this PR been tested?

Tests have been added covering response of controls to selection / insertion / deletion

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

No

## Checklist:

- [X] The code has been tested locally
- [X] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: src/napari_experimental/group_layer_controls.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        icon.setMaximumHeight(50)
+
+        self.layout().addWidget(icon)
+        self.layout().addWidget(label)
+
+
+class QtGroupLayerControlsContainer(QtLayerControlsContainer):
+
+    def __init__(
+        self, viewer: "napari.viewer.Viewer", group_layers: GroupLayer
```
Comment (alessandrofelder) [inline src/napari_experimental/group_layer_controls.py]: Quick docstring here?

File: src/napari_experimental/group_layer_controls.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        Parameters
+        ----------
+        event : Event
+            Event with the target item at `event.value`.
+        """
+        item = event.value
+        self._add_item(item)
+
+    def _add_item(self, item: GroupLayer | GroupLayerNode):
```
Comment (alessandrofelder) [inline src/napari_experimental/group_layer_controls.py]: docstring :)

File: src/napari_experimental/group_layer_controls.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        Parameters
+        ----------
+        event : Event
+            Event with the target item at `event.value`.
+        """
+        item = event.value
+        self._display_item(item)
+
+    def _display_item(self, item: GroupLayer | GroupLayerNode | None):
```
Comment (alessandrofelder) [inline src/napari_experimental/group_layer_controls.py]: docstring :D

**Review by alessandrofelder — CHANGES_REQUESTED** [review]
Thanks @K-Meech - this looks great.

I think I've potentially found a small edge case where things are not wired up exactly as I expect them to, but feel free to disagree with me:

1. make a group layer and a points layer
2. drag the points layer into the group layer
3. expand the group layer
4. the points layer is still "selected" but no points layer controls are shown

I think in 4. I would either expect:
* the points layer to not be selected
* (probably) the group layer to be expanded, the points layer to be selected, and the points layer controls to show

This may be more involved so could also be tackled separately... let me know your thoughts!

(also some nitpicking over docstrings... sorry. Could also be tackled as part of #8 but if we do this frequently #8 will feel overwhelming to tackle :) )

**Comment by K-Meech** [comment]
Thanks @alessandrofelder ! I've updated the docstrings, and added a small fix for your edge case. 

It seems that when items are dragged into a group, it calls [__delitem__](https://github.com/napari/napari/blob/74ecc56c6bf91b1280e6ddca4a54d442cc30d0dd/napari/utils/tree/group.py#L77) followed by [insert](https://github.com/napari/napari/blob/74ecc56c6bf91b1280e6ddca4a54d442cc30d0dd/napari/utils/tree/group.py#L87). `insert` sets the inserted item as the active selection but, as this group layer is nested, it doesn't trigger this line in the controls: `group_layers.selection.events.active.connect(self._display)`. For now, I've added an extra line that listens for changes to the active selection in all `GroupLayer` (including nested ones). I imagine a better solution would be to make sure that any selection changes in nested layers always propagate through to the top `GroupLayer` - this will probably be rather difficult though!

**Review by alessandrofelder — APPROVED** [review]
Thanks @K-Meech - looks good.
Final comment, should probably be addressed separately as this is independent of the layer controls.
Is it intentional that if one reorders an expanded group, it closes? Maybe the state of (expanded/closed) should persist across drag and drop actions?

**Comment by K-Meech** [comment]
@alessandrofelder - I don't think it's intentional. It's something that will be determined by the `QtNodeTreeView` etc that we are inheriting from. I'll make a separate issue for this, and go ahead and merge.

**Pull request merged**